src/lib.rs

```diff
@@ -1232,6 +1232,11 @@ impl EGraph {
         self.proof_state.type_info.sorts.get(&value.tag)
     }
 
+    pub fn add_arcsort(&mut self, arcsort: ArcSort) -> Result<(), TypeError> {
+        self.proof_state.type_info.add_arcsort(arcsort)
+    }
+
+
```

Comment:
Originally I just exposed the `add_sort`, but I needed the more general `add_acrsort` so that I can save the sort in Python.